.github/workflows/release.yaml

```
@@ -6,18 +6,34 @@
 name: Publish Python Package to PyPI
 on:
   push:
-    tags:
-      - '*.*.*'
+    branches:
+      - main
 
 jobs:
   build_and_publish:
     name: Publish a new version of pypdf
     runs-on: ubuntu-latest
+    if: "startsWith(${{ github.event.head_commit.message}}, 'REL: ')"
 
     steps:
       # Ensure it's on PyPI
       - name: Checkout Repository
         uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.PAT }}
+      - name: Extract version from commit message
+        id: extract_version
+        run: |
+          VERSION=$(echo "${{ github.event.head_commit.message }}" | grep -oP '(?<=REL: )\d+\.\d+\.\d+')
+          echo "::set-output name=version::$VERSION"
```

```suggestion
          echo "version=$VERSION" >> $GITHUB_OUTPUT
```

`set-output` is deprecated for quite some time now.